File: lintian_brush/licenses.py
#!/usr/bin/python3
# Copyright (C) 2020 Jelmer Vernooij
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA

"""Utility functions for dealing with licenses."""

import json

from . import open_binary

COMMON_LICENSES_DIR = "/usr/share/common-licenses"

FULL_LICENSE_NAME = {
    "Apache-2.0": "Apache License, Version 2.0",
    "CC0-1.0": "CC0 1.0 Universal license",
}


def load_spdx_data():
    with open_binary("spdx.json") as f:
        return json.load(f)

File: lintian-brush/src/licenses.rs
use std::collections::HashMap;

/// Path to the directory containing common license files on Debian systems
pub const COMMON_LICENSES_DIR: &str = "/usr/share/common-licenses";

lazy_static::lazy_static! {
    /// Mapping of SPDX license identifiers to their full license names
    pub static ref FULL_LICENSE_NAME: HashMap<&'static str, &'static str> = {
        let mut m = HashMap::new();
        m.insert("Apache-2.0", "Apache License, Version 2.0");
        m.insert("CC0-1.0", "CC0 1.0 Universal license");
        m
    };
}
